core/src/main/resources/reference.conf

```diff
@@ -176,13 +176,47 @@ akka.persistence.r2dbc {
 // #connection-settings
 akka.persistence.r2dbc {
 
-  # postgres or yugabyte
+  # postgres, yugabyte or h2
   dialect = "postgres"
 
   # set this to your database schema if applicable, empty by default
   schema = ""
 
-  connection-factory {
+  # Shared defaults, override in the concrete [dialect] you are using
+  default-connection-pool {
+    # Initial pool size.
+    initial-size = 5
+    # Maximum pool size.
+    max-size = 20
+
+    # Maximum idle time of the connection in the pool.
+    # Background eviction interval of idle connections is derived from this property
+    # and max-life-time.
+    max-idle-time = 30 minutes
+
+    # Maximum lifetime of the connection in the pool.
+    # Background eviction interval of connections is derived from this property
+    # and max-idle-time.
+    max-life-time = 60 minutes
+
+    # Maximum time to acquire connection from pool.
+    acquire-timeout = 5 seconds
+    # Number of retries if the connection acquisition attempt fails.
+    # In the case the database server was restarted all connections in the pool will
+    # be invalid. To recover from that without failed acquire you can use the same number
+    # of retries as max-size of the pool
+    acquire-retry = 1
+
+    # Validate the connection when acquired with this SQL.
+    # Enabling this has some performance overhead.
+    # A fast query for Postgres is "SELECT 1"
+    validation-query = ""
+  }
+
+  # dialect specific connection factory configurations,
+  # assign block to 'connection-factory' to chose which one to actually use.
```

Comment:
also see https://doc.akka.io/docs/akka-projection/current/r2dbc.html#multiple-plugins

Reply:
I agree it is a bit weird to have both. Using the dialect name to find the connection factory was one of the things I tried, but dropped that idea to keep backwards compatibility.

Perhaps we should instead pull dialect into the connection factory block since it will always be coupled with that? Then we can keep `akka.persistence.r2dbc.connection-factory` as the way to choose and have predefined blocks for dialects.

Doesn't help me with figuring out how to deal with projections though, with the schema creation needing R2DBCSettings up front.